db/cache_mutation_reader.hh

```diff
@@ -207,6 +210,14 @@ class cache_mutation_reader final : public mutation_reader::impl {
         return gc_clock::time_point::min();
     }
 
+    bool can_gc(tombstone t, is_shadowable is) const {
+        const auto max_purgeable = is ? _max_purgeable_timestamp_shadowable : _max_purgeable_timestamp;
+        if (max_purgeable == api::missing_timestamp) {
+            return false;
+        }
```

Comment:
Unnecessary check (probably eaten by the compiler): missing_timestamp is the lowest possible timestamp, so the check below will always return false.

missing_timestamp is badly named, because it doesn't hint at that. Although, maybe it's not in its contract to also be the lowest possible.

Reply:
Done in v3.